ui/selectors/selectors.js

```diff
@@ -350,22 +364,21 @@ export function getInternalAccount(state, accountId) {
   return state.metamask.internalAccounts.accounts[accountId];
 }
 
-export function getInternalAccountsSortedByKeyring(state) {
-  const accounts = getInternalAccounts(state);
+export const getInternalAccountsSortedByKeyring = createSelector(
+  getMetaMaskKeyrings,
+  getMetaMaskAccounts,
+  (keyrings, accounts) => {
+    // keep existing keyring order
+    const internalAccounts = keyrings
+      .map(({ accounts: addressesOfKeyring }) => addressesOfKeyring)
```

Comment:
IMHO the `addressesOfKeyring` name is a bit confusing, since the scope of this variable is so small we could use just `addresses`.

Reply:
Will address this in a small follow up pr

Reply:
Solved in https://github.com/MetaMask/metamask-extension/pull/23320